File: Problems/42626/Problem.py
def solution(scoville, K):
    answer = -1
    scoville = sorted(scoville)
    for i in range(1, len(scoville)):
        scoville[i] = scoville[i-1] + scoville[i] * 2
        if scoville[i] >= K:
            return i
        del scoville[i]
    return answer

print(solution([1, 2, 3, 9, 10, 12], 7))

File: Problems/12927/Problem.java
import java.util.PriorityQueue;
import java.util.Collections;
import java.util.Arrays;
import java.util.stream.IntStream;

class Problem {
    public long solution(int n, int[] works) {
        long answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>(works.length, Collections.reverseOrder());
        Arrays.stream(works).forEach(work -> pq.offer(work));
        
        IntStream.range(0, n).forEach(hour -> pq.offer(pq.peek() == 0 ? 0 : pq.poll() - 1));
        //pq.stream().forEach(System.out::println);
        
        answer = (long)pq.stream().filter(work -> work != 0).mapToDouble(work -> Math.pow(work.intValue(), 2)).sum();
        
        return answer;
    }

    public static void main(String[] args) {
        System.out.println( new Problem().solution(4, new int[] { 4,3,3 }) );
        System.out.println( new Problem().solution(1, new int[] { 2,1,2 }) );
        System.out.println( new Problem().solution(3, new int[] { 1,1 }) );
    }
}